locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -2742,14 +2743,41 @@ protected LoginCredentials waitForWinRmAvailable(final ComputeService computeSer
             Callable<Boolean> checker = new Callable<Boolean>() {
                 public Boolean call() {
                     for (Map.Entry<WinRmMachineLocation, LoginCredentials> entry : machinesToTry.entrySet()) {
-                        WinRmMachineLocation machine = entry.getKey();
-                        WinRmToolResponse response = machine.executeCommand(
-                                ImmutableMap.of(WinRmTool.PROP_EXEC_TRIES.getName(), 1),
-                                ImmutableList.of("echo testing"));
-                        boolean success = (response.getStatusCode() == 0);
-                        if (success) {
-                            credsSuccessful.set(entry.getValue());
-                            return true;
+                        try {
+                            final WinRmMachineLocation machine = entry.getKey();
+                            WinRmToolResponse response = machine.executeCommand(
+                                    ImmutableMap.of(WinRmTool.PROP_EXEC_TRIES.getName(), 1),
+                                    ImmutableList.of("echo testing"));
+                            boolean success = (response.getStatusCode() == 0);
+                            if (success) {
+                                credsSuccessful.set(entry.getValue());
+                                Predicate<WinRmMachineLocation> pendingReboot = new Predicate<WinRmMachineLocation>() {
```

Comment:
A more proper name would be the reverse - `machineReachable`. It returns true of the command executes successfully.